client/src/context/useSchedulerContext.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useContext, createContext, FunctionComponent, useEffect, useCallback } from 'react';
+
+import moment from 'moment-timezone';
+
+// schedule is meant to store availability from the google calendar api
+// availableDays is meant to store the days and hours available as set by the user from the onboard page
+
+interface ISchedulerContext {
+  selectedDay: Date | undefined;
+  updateSelectedDayContext: (day: Date | undefined) => void;
+  timezone: string | undefined;
+  updateTimezoneContext: (tz: string) => void;
+  schedule: Array<Date> | undefined;
+  updateScheduleContext: (calendar: Array<Date> | undefined) => void;
+  availability: Record<string, Array<string>>;
+  updateAvailabilityContext: (hours: Record<string, Array<string>>) => void;
+  modal: boolean;
+  updateModalContext: (open: boolean) => void;
+  meeting: string;
+  updateMeetingContext: (time: string) => void;
+}
+
+export const SchedulerContext = createContext<ISchedulerContext>({
+  selectedDay: undefined,
+  updateSelectedDayContext: () => null,
+  timezone: undefined,
+  updateTimezoneContext: () => null,
+  schedule: [],
+  updateScheduleContext: () => null,
+  availability: {
+    Sunday: [],
```

Comment:
For my understanding, this will be an array of times they are available on this particular day?

Reply:
So schedule is the array that stores the users current appointments based on what is stored on the calendar api